### PR TITLE
test(pkg/config): migrate test-infra to testify for pkg/config pkg (#2825)

### LINKED ISSUE #2825: migrate test-infra to testify for `pkg/config` pkg

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve? 
migrate test-infra to testify for pkg/config pkg, close #2825 

### What is changed and how it works?
1. migrate test-infra from gocheck to testify
2. migrate leak-test from testleak to goleak

### Check List 

Tests 

 - Unit test

Code changes

- None

Side effects

 - None

Related changes

 - None

### Release note 

```release-note
None
```

### CODE REVIEW

File: pkg/config/config_test.go [hunk 1]
```diff
@@ -16,50 +16,43 @@ package config
...
-func Test(t *testing.T) { check.TestingT(t) }
-
-type replicaConfigSuite struct{}
-
-var _ = check.Suite(&replicaConfigSuite{})
-
-func (s *replicaConfigSuite) TestMarshal(c *check.C) {
-	defer testleak.AfterTest(c)()
+func TestReplicaConfigMarshal(t *testing.T) {
+	t.Parallel()
```
[the remark this answers is not in the document]
Comment (maxshuang) [reply pkg/config/config_test.go:23]: @amyangfei  please take a look at this data race situation,  i  look deep in the config.go but fail to figure out what cause this data race.

[the remark this answers is not in the document]
Comment (maxshuang) [reply pkg/config/config_test.go:23]: @CharlesCheung96 ref to [issue 2957](https://github.com/pingcap/ticdc/issues/2957),  need fix.

**Review by maxshuang — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Pull request merged**